File: src/main/java/com/xellitix/commons/net/url/UrlFactory.java
package com.xellitix.commons.net.url;

import java.net.MalformedURLException;
import java.net.URL;

/**
 * {@link URL} factory.
 *
 * @author Grayson Kuhns
 */
public interface UrlFactory {

  /**
   * Creates a {@link URL}.
   *
   * @param url The {@link URL} string representation.
   * @return The {@link URL}.
   * @throws MalformedURLException If the {@link URL} is malformed.
   */
  URL create(String url) throws MalformedURLException;
}

File: src/main/kotlin/com/xellitix/commons/net/url/UrlFactory.kt
package com.xellitix.commons.net.url

import com.google.inject.Singleton
import java.net.MalformedURLException
import java.net.URL

/**
 * [URL] factory.
 */
@Singleton
class UrlFactory {

    /**
     * Creates a [URL].
     */
    fun create(url: String): URL {
        return try {
            URL(url)
        } catch (ex: MalformedURLException) {
            throw MalformedUrlException(ex)
        }
    }
}
